### ISSUE TITLE
[Paramount+addon] many streams freeze when the ads start.

### ISSUE THREAD

**Issue by micheleciondolo** [matthuisman/slyguy.addons#924]
I’m not sure if this can be considered a bug, if it’s appropriate for me to report it, or anything else.  

The issue is that when I start multiple streams using the Paramount+ plugin, every time an ad starts, Kodi either freezes or even crashes. I’ve tested this on both Debian and Android TV, with the same result.  

For example, if I watch a Champions League match and then the post-match show, every single ad causes the application to freeze. Even if I manage to close the stream in time, I have no way of knowing when the ad ends.  
So, if I were watching a match, I wouldn’t know when it resumes.  

Any ideas on how to fix this (even a workaround)?  
I know the problem might be related to InputStream Adaptive, but do you happen to have any suggestions to avoid the freezing?

**Comment by matthuisman** [comment]
hmm. you could try disable inputstream for hls in the add-on player settings.
See if that helps. that should make it use ffmpeg

**Comment by micheleciondolo** [comment]
Thank you but.. It seems also with ffmepg freezes without crashing the application. Also, by disabling InputStream, I don't think I can watch Serie A.

**Comment by matthuisman** [comment]
It'll still use IA when it's required. But turning off that setting should make it use ffmpeg where possible.

Could you confirm with slyguy proxy disabled that you still have freezes? Just want to make sure that isn't the cause

**Comment by micheleciondolo** [comment]
Yes, it's the same thing. The problem lies in the switch between the main stream, such as a Serie A match, and the advertisements, which apparently use a different encryption or something else. It would be great to find a way to wait for the main stream to resume (even without watching the ads).

**Comment by matthuisman** [comment]
For pluto, we set these extra IA settings
```
inputstream = inputstream.HLS(
    live = True,
    properties = {'manifest_config': '{"hls_ignore_endlist":true,"hls_fix_mediasequence":true,"hls_fix_discsequence":true}'}
),
```
I wonder if they would help with paramount and other addons?

**Comment by micheleciondolo** [comment]
It should be worth a try! Unfortunately, at the moment I only have the Android version of Kodi and I don’t know how to modify this part in Python. If I had a plugin version of this change, I could try it right away.

**Comment by matthuisman** [comment]
ive just pushed a new slyguy common ([script.module.slyguy -> v0.86.44](https://github.com/matthuisman/slyguy.addons/commit/5f571da1bdf5b4dffef99ea450ad8239decadd9e)) that adds those config options to all HLS IA live streams.

**Comment by micheleciondolo** [comment]
Ok. So if I now reinstall Kodi on Android along with your plugin from the repo, I should be able to test it, right? I’ll let you know by tomorrow :)

**Comment by matthuisman** [comment]
yip

**Comment by micheleciondolo** [comment]
Hi Matt. I ran a test but the login to Paramount no longer works from the addon (the official app works perfectly, with various VPNs). It says "failed to fetch json data.. Make sure your IP address is allowed..." but again... it only happens on Kodi. I tried both with a VPN and with DNS. Something must have changed at the login level 😢

**Comment by matthuisman** [comment]
There is a other GitHub issue for the login issue

**Comment by Simone0401** [comment]
I've just tried with the proposal fix but nothing has changed. Live Serie A games suffer freezes, I think due ads

**Comment by micheleciondolo** [comment]
Yes, it keeps having the same error. I put on a CBS Sports live stream, the ad started, then the stream froze... I waited until the ad finished... It started loading the live stream again, freezing several times... In the end, the live had already been running for 1 minute and it was still freezing. The only solution is to close and reopen the stream. Ideally, the ad should play smoothly without any interruption.
